script.js

```diff
@@ -0,0 +1,298 @@
+            break;
+          case 'Ctrl':
+            break;
+          case 'Alt':
+            break;
+          case 'Win':
+            break;
+          default:
+            this.properties.value += keyName;
+            break;
+        }
+        if (keyName !== 'Caps Lock') { setTimeout(() => { key.classList.remove('keyboard--key-active'); }, 300); }
+        this.triggerEvent('oninput');
+      }
+    });
+  }
+
+  createKeys() {
+    this.hardKeys.forEach((keyName) => {
+      const key = document.createElement('div');
+      key.classList.add('keyboard--key');
+      this.elements.keys.push(key);
+      switch (keyName) {
+        case 'Backspace':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'Enter':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'CapsLock':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftLeft':
+          key.classList.add('keyboard--key-wide100');
+          break;
+        case 'ShiftRight':
+          key.classList.add('keyboard--key-wide');
+          break;
+        case 'Delete':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'Tab':
+          key.classList.add('keyboard--key-wider');
+          break;
+        case 'ControlLeft':
+          break;
+        case 'ControlRight':
+          break;
+        case 'Space':
+          key.classList.add('keyboard--key-widest');
+          break;
+        case 'AltLeft':
+          break;
+        case 'AltRight':
+          break;
+        case 'ArrowLeft':
+          break;
+        case 'ArrowRight':
+          break;
+        case 'ArrowUp':
+          break;
+        case 'ArrowDown':
+          break;
+        case 'MetaLeft':
+          break;
+        default:
+          key.classList.add('keyboard--key-forCaps');
+          break;
+      }
+      this.elements.keysContainer.appendChild(key);
+    });
+  }
+
+  triggerEvent(name) {
+    this.eventHandlers[name](this.properties.value);
+  }
+
+  toggleCapsLock() {
+    this.properties.capsLock = !this.properties.capsLock;
+    const currentLayout = this.properties.capsLock ? this.layouts[this.properties.lang].caps
+      : this.layouts[this.properties.lang].nocaps;
+    this.setKeys(currentLayout);
+  }
+
+  switchLang() {
+    if (this.properties.lang === 'en') {
+      this.properties.lang = 'ru';
+      window.sessionStorage.setItem('lang', 'ru');
+    } else {
+      this.properties.lang = 'en';
+      window.sessionStorage.setItem('lang', 'en');
+    }
+    const currentLayout = this.properties.capsLock ? this.layouts[this.properties.lang].caps
+      : this.layouts[this.properties.lang].nocaps;
+    this.setKeys(currentLayout);
+  }
+
+  setKeys(layout) {
+    let i = 0;
+    this.elements.keys.forEach((keyParameter) => {
+      const keyName = layout[i];
+      const key = keyParameter;
+      i += 1;
```

Comment:
forEach method has a second parameter index, so don't need to use `i`, we should use index instead. 
`this.elements.keys.forEach((keyParameter, i)`